fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStoreTest.java

```diff
@@ -694,4 +697,613 @@ public void extensionRegistryWithUnionDescriptor() {
             }
         }
     }
+
+    private void addRecordType(@Nonnull DescriptorProtos.DescriptorProto newRecordType, @Nonnull KeyExpression primaryKey) {
+        metaDataStore.mutateMetaData(metaDataProto -> MetaDataProtoEditor.addRecordType(metaDataProto, newRecordType),
+                recordMetaDataBuilder -> {
+                    recordMetaDataBuilder.getRecordType(newRecordType.getName()).setPrimaryKey(primaryKey);
+                    recordMetaDataBuilder.getRecordType(newRecordType.getName()).setSinceVersion(recordMetaDataBuilder.getVersion());
```

Comment:
It feels awkward to me, now that I look at it, that the user is responsible for setting the since version in this case (i.e., it feels like it should be encapsulated within `addRecordType`).

Reply:
Yeah, I feel the same. Let me check further if there is a reasonable way to get a `Descriptor` from `DescriptorProto`.